include/circt/Dialect/Verif/VerifOps.td

```diff
@@ -212,4 +212,85 @@ def YieldOp : VerifOp<"yield", [
   ];
 }
 
+//===----------------------------------------------------------------------===//
+// Verification intent ops
+//===----------------------------------------------------------------------===//
+
+/// BMC -- short for bounded model check.
+def BMC : I32EnumAttrCase<"BMC", 0, "bmc">;
+/// KIP -- short for k-Inductive-Proof.
+def KIP : I32EnumAttrCase<"KIP", 0, "induction">;
+
+def FormalTestTypeAttr : I32EnumAttr<"FormalTestType", "formal test type",
+                                [BMC, KIP]> {
+  let cppNamespace = "circt::verif";
+}
+
+def FormalOp : VerifOp<"formal"> {
+  let summary = "defines a formal verification test";
+  let description = [{
+    This operation defines a formal verification test with potentially additional
+    assertions and assumptions. This should be written by declaring symbolic 
+    values that are then connected to a hardware instance. These symbols can then
+    be reasoned about in additional assertions. The region will then be converted 
+    into btor2, SMTLib, or SystemVerilog and ignored during regular SystemVerilog 
```

Comment:
Nit: technically the output of an SMT lowering won't be pure SMTLib but a verif op containing SMT dialect - not the end of the world but another phrasing might be helpful in case anyone comes in expecting an exhaustive pure SMTLIB model of their design and can't figure out how to get one.

Reply:
Oh my bad, I will rephrase that